### PR TITLE
 rails-blog-app - Data form

### PR DESCRIPTION
**Pull request by micheaol**
## In this PR: 

- Create forms to perform the following functions:
    - Creates a Post on behalf of the current_user.
    - Create a comment on behalf of the current_user.
    - Allow Users to add likes to Posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,6 +3,11 @@
 Rails.application.routes.draw do
   root 'users#index'
 
+  post 'users/:id/like/create', to: 'likes#create', as: 'like_form'
+  post 'users/:id/comment/create', to: 'comments#create', as: 'comment_form'
```
Comment (rloterh) [inline config/routes.rb:8]: - [ ] Kindly ensure that there is a pair of routes in the `config/routes.rb` file for creating comments like below:

```ruby
   POST   /posts/:post_id/comments(.:format)     comments#create
   GET   /posts/:post_id/comments/new(.:format)     comments#new
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class CommentsController < ApplicationController
```
Comment (rloterh) [inline app/controllers/comments_controller.rb:1]: - [ ] _New_ action for _comments_ is missing. Kindly ensure to define _new_ action for _comments_ controller, similar to:
```ruby
   def new
     @comment = Comment.new
   end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Michael, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (micheaol) [reply config/routes.rb:8]: Hi @rloterh,

Thank you for the feedback. I create the new comment using partials or are you saying I can't do that? Kindly provide more clarification why. Thank you.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (micheaol) [reply app/controllers/comments_controller.rb:1]: @rloterh ,

Thank you for the feedback. You requested changes and you didn't give me any reason why the changes are required. Is the app not working as expected or did you get any error when you try to run migrate?

File: config/routes.rb (hunk 1 above)
Comment (rloterh) [reply config/routes.rb:8]: Hi @micheaol 

It's a nice idea to implement partials in your project. However, this project requires that you create a pair of _routes_ in the `config/routes.rb` file for creating _comments_. Hence the need for _new_ action in _comments_controller.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (rloterh) [reply app/controllers/comments_controller.rb:1]: Hi @micheaol 

It's a nice idea to implement partials in your project. However, this project requires that you create a pair of _routes_ in the `config/routes.rb` file for creating _comments_. Hence the need for _new_ action in _comments_controller.

File: config/routes.rb (hunk 1 above)
Comment (micheaol) [reply config/routes.rb:8]: @rloterh,

The route creating the comments is the /create which right in the routs.rb. Anyways it has been added as requested.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (micheaol) [reply app/controllers/comments_controller.rb:1]: @rloterh ,
Thank you for the feedback. But you never answered any of my questions:

1. Did you get any error messages when you run the app?
2. Is the app not working as expected?

**Review by DammyShittu — APPROVED** [review]
Hi @micheaol, 👋 

## APPROVED 🟢 

Great implementation of all the requirements in this milestone and the changes that were suggested by the previous reviewer. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**